opta/commands/apply.py

```diff
@@ -130,6 +130,42 @@ def _apply(
     else:
         raise Exception(f"Cannot handle upload config for cloud {layer.cloud}")
 
+    service_modules = (
+        layer.get_module_by_type("k8s-service")
+        if layer.cloud == "aws"
+        else layer.get_module_by_type("gcp-k8s-service")
+    )
+
+    if len(service_modules) > 0 and (
+        Terraform.downloaded_state.get(layer.name, False)
+        or Terraform.download_state(layer)
+        and get_cluster_name(layer) is not None
+    ):
+        configure_kubectl(layer)
+
+        for service_module in service_modules:
+            current_tag = current_image_tag(layer)
+            if (
+                current_tag is not None
+                and image_tag is None
+                and service_module.data.get("image", "") == "AUTO"
+                and not test
+            ):
+                response = (
+                    True
+                    if auto_approve
+                    else click.confirm(
+                        f"WARNING There is an existing deployment (tag={current_tag}) and the pods will be killed as you "
+                        f"did not specify an image tag. Would you like to keep the existing deployment alive? (y/n)",
+                    )
+                )
+                if response:
+                    image_tag = current_tag
+                else:
+                    raise RuntimeError("Aborting")
```

Comment:
In the else case, we need to continue the deployment and kill the pods.

Reply:
done